src/lib/edge-functions/proxy.js

```diff
@@ -74,16 +78,27 @@ const initializeProxy = async ({
     projectDir,
   })
   const hasEdgeFunctions = userFunctionsPath !== undefined || internalFunctions.length !== 0
+  let hasServerError = false
 
   return async (req) => {
-    if (req.headers[headers.Passthrough] !== undefined || !hasEdgeFunctions) {
+    if (req.headers[headers.Passthrough] !== undefined || !hasEdgeFunctions || hasServerError) {
       return
     }
 
-    const [geoLocation, { registry }] = await Promise.all([
-      getGeoLocation({ mode: geolocationMode, offline, state }),
-      server,
-    ])
+    let promiseResult
+
+    try {
+      promiseResult = await Promise.all([getGeoLocation({ mode: geolocationMode, offline, state }), server])
+    } catch (error_) {
+      error(error_ instanceof Error ? error_ : format(error_), { exit: false })
```

Comment:
Can you add some context to why we need this check and the `format` call here? The `error` help is [prepared to handle an `Error` object](https://github.com/netlify/cli/blob/main/src/utils/command-helpers.js#L180).